impeller/renderer/backend/metal/gpu_tracer_mtl.mm

```diff
@@ -0,0 +1,60 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <Metal/Metal.h>
+#include "fml/trace_event.h"
+#include "impeller/renderer/backend/metal/context_mtl.h"
+#include "impeller/renderer/backend/metal/formats_mtl.h"
+
+#include <memory>
+
+#include "impeller/renderer/backend/metal/gpu_tracer_mtl.h"
+
+namespace impeller {
+
+void GPUTracerMTL::MarkFrameEnd() {
+#ifdef IMPELLER_DEBUG
```

Comment:
Nit - rather than guarding each method, just don't even allocate this outside of debug?